docs/tutorials/gluon/mnist.md

```diff
@@ -126,71 +125,116 @@ training scope which is defined by `autograd.record()`.
-    # Updates internal evaluation
-    metric.update(label, outputs)
-print('validation acc: %s=%f'%metric.get())
+    # Possibly copy data and labels to the GPU
+    data = batch.data[0].copyto(ctx)
+    labels = batch.label[0].copyto(ctx)
+    metric.update(labels, net(data))
+print('Validaton: {} = {}'.format(*metric.get()))
 assert metric.get()[1] > 0.94
 ```
 
 If everything went well, we should see an accuracy value that is around 0.96, which means that we are able to accurately predict the digit in 96% of test images. This is a pretty good result. But as we will see in the next part of this tutorial, we can do a lot better than that.
 
+That said, a single numer only gives us very limited information on the performance of our neural network. It is always a good idea to actually look at the images on which the network performed poorly, and check for clues on how to improve the performance. We do that with the help of a small function that produces a list of the images where the network got it wrong, together with the predicted and true labels:
+
+```python
+def get_mislabelled(it):
+    """Return list of ``(input, pred_lbl, true_lbl)`` for mislabelled samples."""
+    mislabelled = []
+    it.reset()
+    for batch in it:
+        data = batch.data[0].copyto(ctx)
+        labels = batch.label[0].copyto(ctx)
+        out = net(data)
+        # Predicted label is the index is where the output is maximal
+        preds = nd.argmax(out, axis=1)
+        for d, p, l in zip(data, preds, labels):
+            if p != l:
+                mislabelled.append(
+                    (d.asnumpy(), int(p.asnumpy()), int(l.asnumpy()))
+                )
+    return mislabelled
+```
+
+We can now get the mislabelled images in the training and validation sets and plot a selection of them:
+
+```python
+import numpy as np
+
+sample_size = 8
+wrong_train = get_mislabelled(train_data)
+wrong_val = get_mislabelled(val_data)
+wrong_train_sample = [wrong_train[i] for i in np.random.randint(0, len(wrong_train), size=sample_size)]
+wrong_val_sample = [wrong_val[i] for i in np.random.randint(0, len(wrong_val), size=sample_size)]
+
+import matplotlib.pyplot as plt
+
+fig, axs = plt.subplots(ncols=sample_size)
+for ax, (img, pred, lbl) in zip(axs, wrong_train_sample):
+    fig.set_size_inches(18, 4)
+    fig.suptitle("Sample of wrong predictions in the training set", fontsize=20)
+    ax.imshow(img[0], cmap="gray")
+    ax.set_title("Predicted: {}\nActual: {}".format(pred, lbl))
+    ax.xaxis.set_visible(False)
+    ax.yaxis.set_visible(False)
+
+fig, axs = plt.subplots(ncols=sample_size)
+for ax, (img, pred, lbl) in zip(axs, wrong_val_sample):
+    fig.set_size_inches(18, 4)
+    fig.suptitle("Sample of wrong predictions in the validation set", fontsize=20)
+    ax.imshow(img[0], cmap="gray")
+    ax.set_title("Predicted: {}\nActual: {}".format(pred, lbl))
+    ax.xaxis.set_visible(False)
+    ax.yaxis.set_visible(False)
+```
+![png](./wrong_train.png)
```

Comment:
please host the images on https://github.com/dmlc/web-data in the corresponding path

Reply:
Does that require a separate PR there?

Reply:
Yes indeed, please tag me in and I'll take care of merging it. Try to follow the same path, `mxnet/doc/tutorials/gluon/mnist/your_image.png`

Reply:
Okay, will do.

Reply:
Done, see https://github.com/dmlc/web-data/pull/142